scripts/deploy/deploy.ts

```diff
@@ -47,35 +47,49 @@ export interface DeployedResources {
     vaultFactory: VaultFactory;
 }
 
-export async function main(
-    ORIGINATOR_ROLE = DEFAULT_ORIGINATOR_ROLE,
-    REPAYER_ROLE = DEFAULT_REPAYER_ROLE,
-): Promise<DeployedResources> {
-    // Hardhat always runs the compile task when running scripts through it.
-    // If this runs in a standalone fashion you may want to call compile manually
-    // to make sure everything is compiled
-    // await run("compile");
+// nonce debug toggle
+const DEBUG_NONCE = false;
+
+export async function main(): Promise<DeployedResources> {
+
+    // deployer address from .env
+    let deployerAddr: string;
+    if(process.env.ARCADE_DEPLOYER_ADDRESS) {
+      deployerAddr = process.env.ARCADE_DEPLOYER_ADDRESS;
+    } else {
+      throw new Error("ADDRESS environment variable is not set.")
+    }
+
+    // get deployer accounts current transaction count to create the nonce
+    // upgradeable contracts using deployProxy do have a custom nonce added to them
+    let nonce_counter = await ethers.provider.getTransactionCount(deployerAddr);
 
     console.log(SECTION_SEPARATOR);
+    console.log("Deploying protocol...")
 
+    // ======= CallWhiteList =======
+    console.log(SUBSECTION_SEPARATOR);
+    if (DEBUG_NONCE) console.log("CURRENT NONCE:", nonce_counter);
     const CallWhiteListFactory = await ethers.getContractFactory("CallWhitelist");
-    const whitelist = <CallWhitelist>await CallWhiteListFactory.deploy();
+    const whitelist = <CallWhitelist> await CallWhiteListFactory.deploy({ nonce: nonce_counter });
     await whitelist.deployed();
+    console.log("CallWhiteList deployed to:", whitelist.address);
 
-    const whitelistAddress = whitelist.address;
-    console.log("CallWhiteList deployed to:", whitelistAddress);
+    // ======= AssetVault =======
     console.log(SUBSECTION_SEPARATOR);
-
+    nonce_counter++;
+    if (DEBUG_NONCE) console.log("CURRENT NONCE:", nonce_counter);
     const AssetVaultFactory = await ethers.getContractFactory("AssetVault");
-    const assetVault = <AssetVault>await AssetVaultFactory.deploy();
+    const assetVault = <AssetVault> await AssetVaultFactory.deploy({ nonce: nonce_counter });
     await assetVault.deployed();
+    console.log("AssetVault deployed to:", assetVault.address);
 
-    const assetVaultAddress = assetVault.address;
-    console.log("AssetVault deployed to:", assetVaultAddress);
+    // ======= VaultFactory =======
     console.log(SUBSECTION_SEPARATOR);
-
+    nonce_counter++;
+    if (DEBUG_NONCE) console.log("CURRENT NONCE:", nonce_counter);
     const VaultFactoryFactory = await ethers.getContractFactory("VaultFactory");
-    const vaultFactory = <VaultFactory>await upgrades.deployProxy(
+    const vaultFactory = <VaultFactory> await upgrades.deployProxy(
```

Comment:
Due to this limitation with the OpenZeppelin/Hardhat Upgrades plugin, this is not possible and yes makes it difficult to justify using nonces, since it would only be for a fraction of the deployment calls. I will close this PR since the script on main branch currently has `await <contractName>.deployed()` after all deployments, waiting for the tx to be mined before sending out the next tx /nonce.